### PR TITLE
Add `download` command

### PR DESCRIPTION
**Pull request by hansent**
Adds a `download command`

To be used e.g. like this:
`roboflow download streamlit/streamlit-logos`

or more verbose
`roboflow download -f yolov5pytorch -v 3 streamlit/streamlit-logos`

 - Defaults to latest version and format = `voc` if not specified
 - version can be specified as part of url or as `--version` option (option overwrites url)
 - waits for up to 1 minute for export to generate completely if its not ready yet
 - downloads file to current directory with filename: `${workspaceUrl}-${projectUrl}-${version}-${options.format}.zip`
 - nice progress bar on download

### CODE REVIEW

**Comment by yeldarby** [comment]
> Defaults to latest version and format = voc if not specified

Can we give an interactive selector instead?

**Comment by yeldarby** [comment]
> waits for up to 1 minute for export to generate completely if its not ready yet

What happens after 1 minute?

File: cli/commands/download.js [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    }
+
+    return null;
+}
+
+async function verifyVersionExists(workspaceUrl, projectUrl, version, apiKey) {
+    const availableVersions = await getProjectVersions(workspaceUrl, projectUrl, apiKey);
+    const versionExists = availableVersions.find((v) => v.id.endsWith(`/${version}`));
+    if (!versionExists) {
+        throw new Error(`version ${version} does not exist for dataset ${projectUrl}`);
```
Comment (yeldarby) [inline cli/commands/download.js]: We actually will generate this automatically now! Cc @Jacobsolawetz for the details on that. I think it just auto-zips & returns back a progress amount if not completed so you can keep polling for it.

Comment (hansent) [reply cli/commands/download.js]: Yeah it checks the progress and polls on the export already (for up to a minute)

This is for checking whether the version selected exists (e.g if you try to export/download version 10, but there is only 1,2,3 available)

**Comment by hansent** [comment]
> What happens after 1 minute?

It prints a message saying the server is still generating the export file and asks you to try again in a bit

Could make the timeout configurable, but should maybe increase the polling Intervall after a while

File: cli/commands/download.js [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+async function getDownloadLink(
+    workspaceUrl,
+    projectUrl,
+    version,
+    format,
+    apiKey,
+    recurseDepth = 0
+) {
+    if (recurseDepth > 12) {
```
Comment (yeldarby) [inline cli/commands/download.js]: Let's set this to something higher. Would be cool if we can show a progress bar too.

**Comment by hansent** [comment]
Notes / things left to do from looking at this with @yeldarby:

- [x] handle dataset type (different dataset types support different formats)
- [x] show interactive picker to select format if its not specified as argument
- [x] show a progress bar while version is generating instead of printing messages on each try
- [x] unzip the downloaded file

**Comment by hansent** [comment]
I think this is ready now.  Noticed the API returns error about not supporting format on the `/format` endpoint for e.g. instance segmentation datasets when exporting to `coco-segmentation` or `tfrecrd` although we do support those exports in the UI on universe.  I think it might just be a validation in the newer API endpoint.  Will add a separate fix in the main repo for that.

**Review by yeldarby — APPROVED** [review]

**Pull request merged**